feature/src/common/frame.c

```diff
@@ -219,3 +510,64 @@ int get_frame_offset(const char *fmt, int w, int h, size_t *offset)
     return 0;
 }
 
+int get_color_resolution(enum VmafPixelFormat fmt, unsigned int w, unsigned int h, unsigned int *w_u, unsigned int *h_u, unsigned int *w_v, unsigned int *h_v) {
+
+    // check that both width and height are positive
+    if ((w <= 0) || (h <= 0))
+    {
+        fprintf(stderr, "Width or height is not positive, width = %d, height = %d.\n", w, h);
+        return 1;
+    }
+
+    if (fmt == VMAF_PIX_FMT_YUV420P || fmt == VMAF_PIX_FMT_YUV420P10LE)
```

Comment:
Refactor to use switch? just keep the format consistent with other refactors.

Reply:
Fixed